pkg/resmgr/policy/metrics.go

```diff
@@ -228,11 +196,36 @@ func (s *SystemCollector) Update() {
 	}
 
 	for _, n := range s.Nodes {
-		s.Metrics[nodeUsage].WithLabelValues(n.IdLabel).Set(float64(n.Usage))
+		s.NodeUsage.Record(
+			context.Background(),
+			n.Usage,
+			metric.WithAttributes(
+				attribute.String("node.id", n.IdLabel),
+			),
+		)
+		s.NodeContainers.Record(
+			context.Background(),
+			int64(n.ContainerCount),
+			metric.WithAttributes(
+				attribute.String("node.id", n.IdLabel),
+			),
+		)
 	}
 	for _, c := range s.Cpus {
-		s.Metrics[cpuAllocation].WithLabelValues(c.IdLabel).Set(float64(c.Allocation))
-		s.Metrics[cpuContainers].WithLabelValues(c.IdLabel).Set(float64(c.ContainerCount))
+		s.CpuAllocation.Record(
+			context.Background(),
+			int64(c.Allocation),
+			metric.WithAttributes(
```

Comment:
With a huge number of CPUs, it might be cheaper to use static AttributeSets (metric.WithAttributeSet) to avoid creating so many copies... But I think at this point this might be too early optimization.